apps/src/studio/studio.js

```diff
@@ -3315,8 +3333,8 @@ Studio.setBackground = function (opts) {
 
     // Draw the tiles (again) now that we know which background we're using.
     if (level.wallMapCollisions) {
-      $(".tile_clip").remove();
-      $(".tile").remove();
+      $(".specificTileClip").remove();
```

Comment:
The final commit https://github.com/code-dot-org/code-dot-org/commit/e81386c833d2ef2f9282086eea41f0b655961d79 addresses this issue slightly differently.  Now, the skin doesn't provide a default wall map.  This allows levels which have an explicit wall map (generally levels 7-15) to set that and have it redrawn when background changes etc., while levels that have a random wall map (currently levels 1-6) will render their random tiles once at the beginning, and setMap will be a no-op for them so that they stay rendered for the duration of the level.

It's tempting to consider separating the rendering out into a React-style immediate mode model where we can simplify a bunch of logic like this to not worry about what's already showing.  Maybe after December :)